### DISCUSSION TITLE
How to implement VueFireAppCheck

### DISCUSSION THREAD

**Discussion by Ongomobile** [vuejs/vuefire#1340] [Questions and Help]
I am struggling to figure out how to use Firebase appCheck with VueFire. I set my recaptcha secret in Firebase backend. Note my app is deployed on Netlify.

 I would think the site key needs to be set somewhere that's why I tried use useAppCheckToken & useAppCheck could not get either to work.

This is the error I get: Uncaught TypeError: Cannot read properties of undefined (reading 'debug')

This is what my main.j looks like.

```
<script setup>
import { ref } from 'vue';
import { useFirebaseAuth, useAppCheckToken } from 'vuefire';
import { signInWithEmailAndPassword } from '@firebase/auth';
....

const app = createApp(App)
  .use(VueFire, {
    // imported above but could also just be created here
    firebaseApp,
    modules: [VueFireAuth(), VueFireAppCheck()],
  })
   ......
</script>
```

Here is what I tried in my sign component.

```
<script setup>
import { ref } from 'vue';
import { useFirebaseAuth, useAppCheckToken, useAppCheck } from 'vuefire';
import { signInWithEmailAndPassword } from '@firebase/auth';
import { useRouter } from 'vue-router';
....

const router = useRouter();
const auth = useFirebaseAuth();

const userInput = ref({
  email: '',
  password: '',
});

async function signInToFirebase() {
  // I tried this
  useAppCheck('window-counter-web');
  
// And I tried this
  useAppCheckToken('my site key');

 // I tried this
  useAppCheck('my site key');

 //This method works
  signInWithEmailAndPassword(auth, userInput.value.email, userInput.value.password)
    .then((userCredential) => {
      // Signed in

      const user = userCredential.user;
      router.push('/app/home');
      ...
    })
    .catch((error) => {
      const errorCode = error.code;
      const errorMessage = error.message;
    });
}
</script>
```

**Comment by posva** [comment]
AppCheck should be working properly now. BTW `useAppCheckToken()` (and any other function starting with `use` must be placed at the top level of script setup like other composables)

**Comment by javaboboApp** [comment]
could you please update the documentation and explain how to use it?? https://vuefire.vuejs.org/guide/app-check.html we can not guest the way guys how we need to get the token.. so some examples would be good.
